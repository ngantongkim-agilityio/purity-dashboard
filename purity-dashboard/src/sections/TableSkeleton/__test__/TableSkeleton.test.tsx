import { render } from '@testing-library/react';

// Components
import { TableSkeleton } from '..';

describe('TableSkeleton component', () => {
  it('Should render TableSkeleton component with default props', () => {
    const { container } = render(<TableSkeleton />);

    expect(container).toMatchSnapshot();
  });
});
