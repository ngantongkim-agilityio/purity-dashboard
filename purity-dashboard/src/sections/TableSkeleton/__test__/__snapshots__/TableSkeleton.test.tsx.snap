// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableSkeleton component Should render TableSkeleton component with default props 1`] = `
<div>
  <div
    class="mt-6 flow-root"
  >
    <div
      class="inline-block min-w-full align-middle"
    >
      <div
        class="rounded-lg bg-gray-50 p-2 md:pt-0"
      >
        <div
          class="md:hidden"
        >
          <div
            class="mb-2 w-full rounded-md bg-white p-4"
          >
            <div
              class="flex items-center justify-between border-b border-gray-100 pb-8"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="mr-2 h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </div>
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </div>
            <div
              class="flex w-full items-center justify-between pt-4"
            >
              <div>
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
                <div
                  class="mt-2 h-6 w-24 rounded bg-gray-100"
                />
              </div>
              <div
                class="flex justify-end gap-2"
              >
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
              </div>
            </div>
          </div>
          <div
            class="mb-2 w-full rounded-md bg-white p-4"
          >
            <div
              class="flex items-center justify-between border-b border-gray-100 pb-8"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="mr-2 h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </div>
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </div>
            <div
              class="flex w-full items-center justify-between pt-4"
            >
              <div>
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
                <div
                  class="mt-2 h-6 w-24 rounded bg-gray-100"
                />
              </div>
              <div
                class="flex justify-end gap-2"
              >
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
              </div>
            </div>
          </div>
          <div
            class="mb-2 w-full rounded-md bg-white p-4"
          >
            <div
              class="flex items-center justify-between border-b border-gray-100 pb-8"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="mr-2 h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </div>
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </div>
            <div
              class="flex w-full items-center justify-between pt-4"
            >
              <div>
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
                <div
                  class="mt-2 h-6 w-24 rounded bg-gray-100"
                />
              </div>
              <div
                class="flex justify-end gap-2"
              >
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
              </div>
            </div>
          </div>
          <div
            class="mb-2 w-full rounded-md bg-white p-4"
          >
            <div
              class="flex items-center justify-between border-b border-gray-100 pb-8"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="mr-2 h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </div>
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </div>
            <div
              class="flex w-full items-center justify-between pt-4"
            >
              <div>
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
                <div
                  class="mt-2 h-6 w-24 rounded bg-gray-100"
                />
              </div>
              <div
                class="flex justify-end gap-2"
              >
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
              </div>
            </div>
          </div>
          <div
            class="mb-2 w-full rounded-md bg-white p-4"
          >
            <div
              class="flex items-center justify-between border-b border-gray-100 pb-8"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="mr-2 h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </div>
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </div>
            <div
              class="flex w-full items-center justify-between pt-4"
            >
              <div>
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
                <div
                  class="mt-2 h-6 w-24 rounded bg-gray-100"
                />
              </div>
              <div
                class="flex justify-end gap-2"
              >
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
              </div>
            </div>
          </div>
          <div
            class="mb-2 w-full rounded-md bg-white p-4"
          >
            <div
              class="flex items-center justify-between border-b border-gray-100 pb-8"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="mr-2 h-8 w-8 rounded-full bg-gray-100"
                />
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </div>
              <div
                class="h-6 w-16 rounded bg-gray-100"
              />
            </div>
            <div
              class="flex w-full items-center justify-between pt-4"
            >
              <div>
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
                <div
                  class="mt-2 h-6 w-24 rounded bg-gray-100"
                />
              </div>
              <div
                class="flex justify-end gap-2"
              >
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
                <div
                  class="h-10 w-10 rounded bg-gray-100"
                />
              </div>
            </div>
          </div>
        </div>
        <table
          class="hidden min-w-full text-gray-900 md:table"
        >
          <thead
            class="rounded-lg text-left text-sm font-normal"
          >
            <tr>
              <th
                class="px-4 py-5 font-medium sm:pl-6"
                scope="col"
              >
                Customer
              </th>
              <th
                class="px-3 py-5 font-medium"
                scope="col"
              >
                Email
              </th>
              <th
                class="px-3 py-5 font-medium"
                scope="col"
              >
                Amount
              </th>
              <th
                class="px-3 py-5 font-medium"
                scope="col"
              >
                Date
              </th>
              <th
                class="px-3 py-5 font-medium"
                scope="col"
              >
                Status
              </th>
              <th
                class="relative pb-4 pl-3 pr-6 pt-2 sm:pr-6"
                scope="col"
              >
                <span
                  class="sr-only"
                >
                  Edit
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="bg-white"
          >
            <tr
              class="w-full border-b border-gray-100 last-of-type:border-none [&:first-child>td:first-child]:rounded-tl-lg [&:first-child>td:last-child]:rounded-tr-lg [&:last-child>td:first-child]:rounded-bl-lg [&:last-child>td:last-child]:rounded-br-lg"
            >
              <td
                class="relative overflow-hidden whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex items-center gap-3"
                >
                  <div
                    class="h-8 w-8 rounded-full bg-gray-100"
                  />
                  <div
                    class="h-6 w-24 rounded bg-gray-100"
                  />
                </div>
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-32 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex justify-end gap-3"
                >
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                </div>
              </td>
            </tr>
            <tr
              class="w-full border-b border-gray-100 last-of-type:border-none [&:first-child>td:first-child]:rounded-tl-lg [&:first-child>td:last-child]:rounded-tr-lg [&:last-child>td:first-child]:rounded-bl-lg [&:last-child>td:last-child]:rounded-br-lg"
            >
              <td
                class="relative overflow-hidden whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex items-center gap-3"
                >
                  <div
                    class="h-8 w-8 rounded-full bg-gray-100"
                  />
                  <div
                    class="h-6 w-24 rounded bg-gray-100"
                  />
                </div>
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-32 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex justify-end gap-3"
                >
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                </div>
              </td>
            </tr>
            <tr
              class="w-full border-b border-gray-100 last-of-type:border-none [&:first-child>td:first-child]:rounded-tl-lg [&:first-child>td:last-child]:rounded-tr-lg [&:last-child>td:first-child]:rounded-bl-lg [&:last-child>td:last-child]:rounded-br-lg"
            >
              <td
                class="relative overflow-hidden whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex items-center gap-3"
                >
                  <div
                    class="h-8 w-8 rounded-full bg-gray-100"
                  />
                  <div
                    class="h-6 w-24 rounded bg-gray-100"
                  />
                </div>
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-32 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex justify-end gap-3"
                >
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                </div>
              </td>
            </tr>
            <tr
              class="w-full border-b border-gray-100 last-of-type:border-none [&:first-child>td:first-child]:rounded-tl-lg [&:first-child>td:last-child]:rounded-tr-lg [&:last-child>td:first-child]:rounded-bl-lg [&:last-child>td:last-child]:rounded-br-lg"
            >
              <td
                class="relative overflow-hidden whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex items-center gap-3"
                >
                  <div
                    class="h-8 w-8 rounded-full bg-gray-100"
                  />
                  <div
                    class="h-6 w-24 rounded bg-gray-100"
                  />
                </div>
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-32 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex justify-end gap-3"
                >
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                </div>
              </td>
            </tr>
            <tr
              class="w-full border-b border-gray-100 last-of-type:border-none [&:first-child>td:first-child]:rounded-tl-lg [&:first-child>td:last-child]:rounded-tr-lg [&:last-child>td:first-child]:rounded-bl-lg [&:last-child>td:last-child]:rounded-br-lg"
            >
              <td
                class="relative overflow-hidden whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex items-center gap-3"
                >
                  <div
                    class="h-8 w-8 rounded-full bg-gray-100"
                  />
                  <div
                    class="h-6 w-24 rounded bg-gray-100"
                  />
                </div>
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-32 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex justify-end gap-3"
                >
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                </div>
              </td>
            </tr>
            <tr
              class="w-full border-b border-gray-100 last-of-type:border-none [&:first-child>td:first-child]:rounded-tl-lg [&:first-child>td:last-child]:rounded-tr-lg [&:last-child>td:first-child]:rounded-bl-lg [&:last-child>td:last-child]:rounded-br-lg"
            >
              <td
                class="relative overflow-hidden whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex items-center gap-3"
                >
                  <div
                    class="h-8 w-8 rounded-full bg-gray-100"
                  />
                  <div
                    class="h-6 w-24 rounded bg-gray-100"
                  />
                </div>
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-32 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap px-3 py-3"
              >
                <div
                  class="h-6 w-16 rounded bg-gray-100"
                />
              </td>
              <td
                class="whitespace-nowrap py-3 pl-6 pr-3"
              >
                <div
                  class="flex justify-end gap-3"
                >
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                  <div
                    class="h-[38px] w-[38px] rounded bg-gray-100"
                  />
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
